src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -367,6 +372,23 @@ private static string GetPreferredUsernameFromIdToken(bool isAdfsAuthority, IdTo
 
             CacheInfoTelemetry cacheInfoTelemetry = CacheInfoTelemetry.None;
 
+            // Verify appropriate scenario for OBO tokens in long running process
+            if (requestParams.ApiId == ApiEvent.ApiIds.AcquireTokenOnBehalfOf)
+            {
+                // Validates correct AcquireTokenInLongRunningProcess call (user assertion is null in this case)
+                if (requestParams.UserAssertion == null && !string.IsNullOrEmpty(requestParams.OboCacheKey) && tokenCacheItems.Count == 0)
+                {
+                    logger.Error("[FindAccessTokenAsync] AcquireTokenInLongRunningProcess was called and OBO token was not found in the cache.");
+                    throw new MsalClientException(MsalError.OboCacheKeyNotInCacheError, MsalErrorMessage.OboCacheKeyNotInCache);
```

Comment:
Shouldn't this throw an `MsalUiRequiredException` instead? CC @jmprieur for exception behavior.

Reply:
Im not sure it needs to, it may be that the developer simply hasnt supplied the already acquired userAssertion using `InitiateLongRunningProcess` right?